core/app/models/spree/order.rb

```
@@ -907,5 +910,38 @@
+        context: { event_name: 'order_completed', record: {order: self} },
+        source: 'spree.core'
+      )
+    end
+
+    def track_order_cancelled_event
+      analytics_event_handlers = Spree::Analytics.event_handlers.map do |handler|
+        handler.new(user: user, session: nil, request: nil, store: store)
+      end
+
+
+      analytics_event_handlers.each do |handler|
```

RuboCop (1.73)

[convention] Extra blank line detected.

(Layout/EmptyLines)